### ISSUE TITLE
Question for reproducing Table 2

### ISSUE THREAD

**Issue by gihyunkim** [YilinLiu97/FasterDIP-devil-in-upsampling#4]
Hi, I am very interested in your paper and experiments and I want to reproduce Table2 for Set9 Dataset.

I have one question. Actually, I already reproduced original deep image prior code for denoising and I tested it into  Set9. However, I got only 29 PSNR and 0.85 SSIM, worse than the result in Table2. 

Would you explain what is different from DIP setting on your experiment?

Thank you.

**Comment by gihyunkim** [comment]
Also, what is difference between denoising.py and train_unsup_morph.py? 

And your model is DD.py and the script for reproducing your model is denoising_dd_BilinearUp.sh?

**Comment by YilinLiu97** [comment]
Hi @gihyunkim, thanks for your interest and sorry for the late response.  When using DIP-like methods, please check these subtle aspects: 1) the inputs: whether it's from an uniform or Gaussian distribution, and the corresponding hyperparameters (e.g., mu, sigma). This could critically influence the results and the speed of overfitting, sometimes regardless of the rest of the architectural components (discussed in more detail in our new ECCV paper). 2) the upsampling type. Nearest neighbor fits faster than bilinear.  3) the random noise added to the input, i.e., "reg_noise_std", please set it properly according to different noise levels.  

As I have not fully cleaned up the code, please ignore the "train_unsup_morph.py".  We did not propose a particular type of architecture but presented rationales for more principled architectural tuning for DIP. The architecture we used as comparisons with DIP in the experiments is also an encoder-decoder architecture.

**Comment by gihyunkim** [comment]
Thanks for your answer@YilinLiu97. I understand the aspect of DIP-like methods (three things above), however, I could have not reproduced the score for original DIP in table 2 of your paper, although I have not changed any of the option from the default setting. 

Would you check my setting for your code below and let me know how can I reproduce it? (only --decay and --reg_noise_std are different from the default config)

============================================================================

python -u denoising.py --special DIP_orig --task denoising --folder_path ../data/denoise/set9 --model_type DIP_orig --num_iters 3000 --norm_func bn --decay 1e-5 0 --reg_noise_std '1/30' --lr 0.01 --noise_type u --noise_method noise

============================================================================

it gaves me average 30.23 PSNR and only 0.86 SSIM (0.90 in table2) for Set9 dataset.

**also I use ssim metric from piq module since the compare_ssim doesn't work in my environment (it shows much lower ssim than piq).

Thank you.

**Comment by gihyunkim** [comment]
Find out that decay set 1e-7 or 1e-5 in the script is not applied since the args.reg_type=0 and args.sr=0. Could be this problem?

**I tried and it was not the problem.
